app/assets/javascripts/search.js

```diff
@@ -1,11 +1,13 @@
 (function() {
-  var delay, ready;
+  var debounce, ready;
 
-  delay = (function() {
+  debounce = (function() {
     var timer;
     timer = 0;
     return function(callback, ms) {
-      clearTimeout(timer);
+      if(timer) {
```

Comment:
Nitpick: add extra spacing after `if`
```suggestion
      if (timer) {
```